File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        TreeNode current = root;
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(current);
        while(!q.isEmpty()) {
            current = q.poll();
            if(current.right != null) q.add(current.right);
            if(current.left != null) q.add(current.left);
        }
        return current.val;
    }
}

File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        TreeNode *current = root;
        queue<TreeNode*> q;
        q.emplace(current);
        while(!q.empty()) {
            current = q.front();
            q.pop();
            if(current->right) q.emplace(current->right);
            if(current->left) q.emplace(current->left);
        }
        return current->val;
    }
};